jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/SubscribeSnsDialog.kt

```diff
@@ -87,6 +110,32 @@ class SubscribeSnsDialog(
         }
     }
 
+    private fun topicSelected(): String = view.topicSelector.selected()?.topicArn() ?: ""
+
+    private fun needToEditPolicy(existingPolicy: String?): Boolean {
+        existingPolicy ?: return true
+
+        // This does not actually make API calls
+        val allowed = iamClient.simulateCustomPolicy {
+            it.contextEntries(
+                ContextEntry.builder()
+                    .contextKeyType(ContextKeyTypeEnum.STRING)
+                    .contextKeyName("aws:SourceArn")
+                    .contextKeyValues(topicSelected())
+                    .build()
+            )
+            it.actionNames("sqs:SendMessage")
+            it.resourceArns(queue.arn)
+            it.policyInputList(existingPolicy)
+        }.evaluationResults().first()
+
+        if (allowed.evalDecision() != PolicyEvaluationDecisionType.ALLOWED) {
```

Comment:
could be simplified to `return allowed.evalDecision() != PolicyEvaluationDecisionType.ALLOWED`

Reply:
True, fixed